src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -351,6 +355,33 @@ class Config:
         return await self.middleware.call('datastore.delete', self._config.datastore, id)
 
 
+def _run_command(user, commandline, q, rv):
+    setusercontext(user)
+
+    os.environ['PATH'] = (
+        '/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin:/root/bin'
+    )
+    proc = subprocess.Popen(
+        commandline, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
+    )
+
+    # logger(1) does not honor original exit value so we use syslog module instead
+    syslog.openlog(ident='rsync')
```

Comment:
Are we going to make this obsolete at some point?

Reply:
TBH, i am just moving code around, not reimplementing things nor changing behavior.